.taskcluster.yml

```diff
@@ -6,259 +6,212 @@ version: 1
-        expires_in: {$fromNow: '1 year'}
-        scheduler_id: focus-nightly-sched
-        user: ${event.sender.login}
-        repository: ${event.repository.clone_url}
-      in:
+  $let:
+    decision_task_id: {$eval: as_slugid("decision_task")}
+    expires_in: {$fromNow: '1 year'}
+    user: ${event.sender.login}
+
+    # We define the following variable at the very top, because they are used in the
+    # default definition
+    head_branch:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.ref}
+      else:
+        $if: 'tasks_for == "github-push"'
+        then: ${event.ref}
+        else: ${event.release.target_commitish}
+
+    head_rev:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.sha}
+      else:
+        $if: 'tasks_for == "github-push"'
+        then: ${event.after}
+        else: ${event.release.tag_name}
+
+    repository:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.repo.html_url}
+      else: ${event.repository.html_url}
+
+    worker_type:
+      $if: 'tasks_for in ["github-push", "github-release"] && event.repository.html_url == "https://github.com/mozilla-mobile/android-components"'
+      then: mobile-3-decision
+      else: mobile-1-decision
+
+    scheduler_id:
+      $if: 'tasks_for == "cron"'
+      then: focus-nightly-sched  # TODO: Rename to mobile-nightly-sched
+      else: taskcluster-github
+
+    does_event_occur_on_main_repo:
+      $if: 'event.repository.html_url == "https://github.com/mozilla-mobile/android-components"'
+      then: true
+      else: false
+  in:
+    $let:
+      default_task_definition:
         taskId: ${decision_task_id}
-        taskGroupId: ${decision_task_id}
+        taskGroupId: ${decision_task_id}  # Must be explicit because of Chain of Trust
+        schedulerId: ${scheduler_id}
         created: {$fromNow: ''}
         deadline: {$fromNow: '2 hours'}
         expires: ${expires_in}
-        schedulerId: ${scheduler_id}
         provisionerId: aws-provisioner-v1
-        workerType: gecko-focus
-        requires: all-completed
-        priority: medium
+        workerType: ${worker_type}
+        requires: all-completed   # Must be explicit because of Chain of Trust
         retries: 5
         scopes:
-          - project:mobile:android-components:releng:beetmover:action:push-to-maven
-          - project:mobile:android-components:releng:beetmover:bucket:maven-snapshot-staging
-          - queue:create-task:highest:scriptworker-prov-v1/mobile-beetmover-dev
-          - queue:create-task:highest:aws-provisioner-v1/gecko-focus
-          - queue:create-task:highest:aws-provisioner-v1/github-worker
           - queue:scheduler-id:${scheduler_id}
         payload:
-          maxRunTime: 600
+          maxRunTime: 600   # Decision should remain fast enough to schedule a handful of tasks
           image: mozillamobile/android-components:1.11
```

Comment:
Oh wow, is it `11` already? Nice.

Reply:
Just updated today :)

Reply:
Freshly baked ~3h ago, found it eventually in dockerhub, nice :)